Dailymotion Bid Adaptor: initial release

kvnsw opened the pull request:
<!--
Thank you for your pull request! 

Please title your pull request like this: 'Module: Change', eg 'Fraggles Bid Adapter: support fragglerock'

Please make sure this PR is scoped to one change or you may be asked to resubmit. 
 
Please make sure any added or changed code includes tests with greater than 80% code coverage. 

See https://github.com/prebid/Prebid.js/blob/master/CONTRIBUTING.md#testing-prebidjs for documentation on testing Prebid.js.

For any user facing change, submit a link to a PR on the docs repo at https://github.com/prebid/prebid.github.io/
-->

## Type of change
<!-- Remove items that don't apply and/or select an item by changing [ ] to [x] -->
- [x] New bidder adapter : docs PR (prebid.github.io) -> https://github.com/prebid/prebid.github.io/pull/5091

## Description of change

A new bidder adapter dailymotionBidAdapter added. 

- contact email of the adapter’s maintainer : **ad-leo-engineering@dailymotion.com**

- Test parameters for validating bids:

By setting the following configuration options, you'll get a constant response to any request to validate your adapter integration:

```javascript
const adUnits = [
  {
    bids: [{
      bidder: 'dailymotion',
      params: {
        apiKey: 'dailymotion-testing'
      }
    }],
    code: 'test-ad-unit',
    mediaTypes: {
      video: {
        api: [2, 7],
        context: 'instream',
        playerSize: [ [1280, 720] ],
        startDelay: 0
      },
    }
  }
];
```

<!-- For new bidder adapters, please provide the following
Be sure to test the integration with your adserver using the [Hello World](/integrationExamples/gpt/hello_world.html) sample page. -->

dgirardi commented on modules/dailymotionBidAdapter.js:
```diff
@@ -0,0 +1,55 @@
+    method: 'POST',
+    url: 'https://pb.dmxleo.com',
+    data: {
+      bidder_request: bidderRequest,
```
FYI, the structure of these request objects is not stable; major versions very often introduce breaking changes to them, and sometimes even minor updates do if the refactoring required is small enough. 

People are also not required to keep Prebid up to date so as those changes happen, you'd need not only to immediately update your server side logic but also somehow keep every previous version working.

dgirardi replied:
I realize this is probably coming too late - but if your goal is to do as much as possible server side, my recommendation is to use ortb, which is an actual protocol. The client can be [about as simple](https://github.com/prebid/Prebid.js/tree/master/libraries/ortbConverter), and while I'm not familiar with server side tooling, it's an established standard so I'd expect to find some support there as well.

kvnsw replied:
Hello @dgirardi, we can maybe continue the discussion we started on https://github.com/dailymotion-oss/Prebid.js/pull/4#discussion_r1474155769 here with the updated PR?

kvnsw commented:
Hello @jlquaccia, following changes related to @dgirardi's feedback, could we have a second review? :) Thanks!

patmmccann commented on modules/dailymotionBidAdapter.md:
````diff
@@ -0,0 +1,74 @@
+Before calling this adapter, you need to set its configuration with a call to setConfig like this:
+
+```javascript
+config.setConfig({
````
Don't configure your adapter this way, see how other bidders define params on the ad unit, eg the appnexus md file, or using pbjs.setBidderConfig, eg on grid.

kvnsw replied:
Hello @patmmccann thank you for the feedback! Updated to use `setBidderConfig` on latest commit and updated documentation on https://github.com/prebid/prebid.github.io/pull/5091 as well

patmmccann commented on modules/dailymotionBidAdapter.js:
```diff
@@ -0,0 +1,75 @@
+   * @return boolean True if this is a valid bid, and false otherwise.
+   */
+  isBidRequestValid: () => {
+    const dmConfig = config.getConfig('dailymotion');
```
as noted below; don't call getConfig

sebmil-daily replied:
Hello @ChrisHuie, we are not using `getConfig` for ad unit params anymore as recommended here https://github.com/prebid/Prebid.js/pull/10970#discussion_r1501351540.

Also, we have updated our documentation to specify that the config should be set using `pbjs.setBidderConfig`.

Is there a better way to retrieve the parameters set by the call to `pbjs.setBidderConfig`?

I saw that this type of call is already used in other adapters like adtarget or appier.

kvnsw replied:
Hello @patmmccann, @dgirardi, @ChrisHuie, following's @sebmil-daily last week, could we have some additional informations on why `getConfig` is a blocker for this to move forward please?

dgirardi replied:
In this case it's for consistency, publishers are used to provide this sort of data through [bid parameters](https://docs.prebid.org/dev-docs/bidder-adaptor.html#bidder-adaptor-Designing-your-Bid-Params). An example of an adapter that takes an API key: [revcontent](https://github.com/prebid/Prebid.js/blob/c27f64b1e059c83706435db0b1eeffc88e9b3559/modules/revcontentBidAdapter.js#L33-L35)

kvnsw replied:
Thanks @dgirardi, we'll change it now

kvnsw replied:
@dgirardi we are now using bid params. Let us know if this works for you

kvnsw commented:
Hello @patmmccann, @dgirardi, @ChrisHuie, sorry to bother, I think we have addressed all your feedbacks now and have a final submission. Please let us know if we can move forward :) Thank you!

patmmccann commented on modules/dailymotionBidAdapter.js:
```diff
@@ -0,0 +1,112 @@
+        refererInfo: {
+          page: bidderRequest?.refererInfo?.page || '',
+        },
+        uspConsent: bidderRequest?.uspConsent || '',
```
you should pass yourself the gpp string as usp is past end of life

patmmccann commented on modules/dailymotionBidAdapter.js:
```diff
@@ -0,0 +1,112 @@
+      config: {
+        api_key: bid.params.apiKey
+      },
+      coppa: config.getConfig('coppa'),
```
please get this from the request and not the confgi

patmmccann commented on modules/dailymotionBidAdapter.js:
```diff
@@ -0,0 +1,112 @@
+        api_key: bid.params.apiKey
+      },
+      coppa: config.getConfig('coppa'),
+      request: {
```
This request is quite sparse in comparison to your competitors. Why don't you want the publisher floor, the eids, or the user.data object? You also might find site.content.id will work for you as up to three vendors may be trying to set it in a race and you need the site.content ecids.

patmmccann commented:
@kvnsw taking over this review from Karim; looking forward to working with you. Feel free to slack me

karimMourra commented:
@patmmccann Note that the Video Taskforce had some concerns about this PR:

In an effort to simplify configuration for publishers, the Prebid Video Taskforce is requiring bid adapters to follow the oRTB spec. 

- `mediaTypes.video` should contain fields from oRTB section 3.2.7 

- For content specific fields, we encourage reading from `site.content`, see ortb 2.6 section 3.2.13

Reference: ortb 2.6 spec: https://iabtechlab.com/wp-content/uploads/2022/04/OpenRTB-2-6_FINAL.pdf](https://github.com/prebid/Prebid.js/pull/10970)

anything outside of the spec can be included in the bidder params (i.e. params.video). Though the implementation wouldn't change, we are concerned with the examples and documentation mentioning non-ortb fields in `mediaTypes.video`

m-gendi commented:
Thank you @patmmccann and @karimMourra we appreciate it.
We will address the remaining comments in a new PR, aiming to resolve them.
